lxd/network/driver_ovn.go

```diff
@@ -1166,6 +1166,78 @@ func (n *ovn) getLogicalRouterPeerPortName(peerNetworkID int64) openvswitch.OVNR
 	return openvswitch.OVNRouterPort(fmt.Sprintf("%s-lrp-peer-net%d", n.getRouterName(), peerNetworkID))
 }
 
+// getUplinkGatewayUsage returns information about usage of external subnets by the uplink's gateways.
+func (n *ovn) getUplinkGatewayUsage() ([]externalSubnetUsage, error) {
+	var uplinkNetwork *api.Network
+	var err error
+
+	uplinkName := n.config["network"]
+	if uplinkName == "" {
+		return nil, fmt.Errorf(`OVN network %q is missing "network" config option`, n.name)
+	}
+
+	err = n.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+		_, uplinkNetwork, _, err = tx.GetNetworkInAnyState(ctx, api.ProjectDefaultName, uplinkName)
+		return err
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Failed to load uplink network: %w", err)
+	}
+
+	if uplinkNetwork == nil {
+		return nil, errors.New("Failed to load uplink network")
+	}
+
+	uplinkPhysicalIP4GatewayKey := "ipv4.gateway"        // Uses CIDR notation.
+	uplinkPhysicalIP6GatewayKey := "ipv6.gateway"        // Uses CIDR notation.
+	uplinkBridgeIP4AddrKey := "ipv4.address"             // Uses CIDR notation.
+	uplinkBridgeIP6AddrKey := "ipv6.address"             // Uses CIDR notation.
+	uplinkBridgeIP4DHCPGatewayKey := "ipv4.dhcp.gateway" // Uses bare IP.
```

Comment:
ovn networks dont have this option https://documentation.ubuntu.com/lxd/latest/reference/network_ovn/#configuration-options

Reply:
Oh I see, this is the uplink keys ofc, so the uplink can be a managed bridge and have this option. All good.